core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/impl/ProjectImpl.java

```diff
@@ -61,4 +154,317 @@ public ProjectImpl(User user) {
     setSecurityProfile(new SecurityProfile(user));
   }
 
+  @Override
+  public Date getCreationDate() {
+    return creationDate;
+  }
+
+  @Override
+  public String getDescription() {
+    return description;
+  }
+
+  @Override
+  public String getName() {
+    return name;
+  }
+
+  @Override
+  public String getAlias() {
+    return alias;
+  }
+
+  @Override
+  public String getShortName() {
+    return shortName;
+  }
+
+  @Override
+  public void setShortName(String shortName) {
+    this.shortName = shortName;
+  }
+
+  @Override
+  public long getId() {
+    return projectId;
+  }
+
+  @Override
+  public void setId(long id) {
+    this.projectId = id;
+  }
+
+  @Override
+  @Deprecated
+  public Long getProjectId() {
+    return projectId;
+  }
+
+  @Override
+  public Collection<Sample> getSamples() {
+    return samples;
+  }
+
+  @Override
+  public Collection<Study> getStudies() {
+    return studies;
+  }
+
+  @Override
+  public Collection<ProjectOverview> getOverviews() {
+    return overviews;
+  }
+
+  @Override
+  public ProjectOverview getOverviewById(Long overviewId) {
+    for (ProjectOverview p : getOverviews()) {
+      if (p.getId() == overviewId) {
+        return p;
+      }
+    }
+    return null;
+  }
+
+  @Override
+  public void setCreationDate(Date date) {
+    this.creationDate = date;
+  }
+
+  @Override
+  public void setDescription(String description) {
+    this.description = description;
+  }
+
+  @Override
+  public void setName(String name) {
+    this.name = name;
+  }
+
+  @Override
+  public void setAlias(String alias) {
+    this.alias = alias;
+  }
+
+  @Override
+  @Deprecated
+  public void setProjectId(Long projectId) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Add the missing @deprecated Javadoc tag. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AMissingDeprecatedCheck)
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Do not forget to remove this deprecated code someday. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1133)